jdi-light/src/main/java/com/epam/jdi/light/elements/complex/table/DataGrid.java

```diff
@@ -0,0 +1,214 @@
+        return IDataGrid.super.count();
+    }
+    @Override
+    public WebList webRow(int rowNum) {
+        return IDataGrid.super.webRow(rowNum);
+    }
+    @Override
+    public WebList webColumn(int colNum) {
+        return IDataGrid.super.webColumn(colNum);
+    }
+    @Override
+    public WebList webColumn(String colName) {
+        return IDataGrid.super.webColumn(colName);
+    }
+    @Override
+    public UIElement webCell(int colNum, int rowNum) {
+        return IDataGrid.super.webCell(colNum, rowNum);
+    }
+    @Override
+    public List<String> header() {
+        return IDataGrid.super.header();
+    }
+    @Override
+    public WebList footerUI() {
+        return IDataGrid.super.footerUI();
+    }
+
+    public D rowAsData(WebList row) {
+        return lineClass != null
+            ? lineToData(rowAsLine(row))
+            : new Line(header(), row).asData(dataClass);
+    }
+    public L rowAsLine(WebList row) {
+        return new Line(header(), row).asLine(lineClass);
+    }
+    public List<D> elements(int minAmount) {
+        return allData();
+    }
+    public D get(String value) {
+        return data(value);
+    }
+    @Override
+    public WebList webCells() {
+        return grid().webCells();
+    }
+    @Override
+    public void clear() {
+        grid().clear();
+    }
+    @Override
+    public WebList headerUI() {
+        WebList headerUI = grid().headerUI();
+        return headerUI.size() == grid().size
+            ? headerUI
+            : tryFilterHeader(headerUI);
+    }
+
+    @Override
+    public IDataGridAssert<D, IDataGrid<L, D>,?> is() {
+        return new IDataGridAssert<>(this);
+    }
+
+    public void setup(Field field) {
+        grid().setup(field);
+        try {
+            setupGenericFields(field);
+        } catch (Exception ignore) { // ignore if can't setup
+        }
+        List<Field> entityFields = new ArrayList<>();
+        if (lineClass != null)
+            entityFields.addAll(getFieldsExact(lineClass, f -> isInterface(f, HasValue.class)));
+        if (dataClass != null)
+            entityFields.addAll(asList(dataClass.getDeclaredFields()));
+        if (entityFields.size() > 0) {
+            grid().header = LinqUtils.map(entityFields, field1 -> splitCamelCase(field1.getName()))
+                .stream().distinct().collect(Collectors.toList());
+            grid().size = grid().header.size();
+        }
+    }
+
+    protected void setupGenericFields(Field field) {
+        Type[] types = getGenericTypes(field);
+        if (types.length != 2)
+            return;
+        try {
+            lineClass = types[0].toString().equals("?") ? null : (Class<L>) types[0];
+            dataClass = types[1].toString().equals("?") ? null : (Class<D>) types[1];
+        } catch (Exception ex) {
+            throw exception(ex, "Can't get DataTable %s data or entity class", getName());
+        }
+    }
+
+    public JFunc1<String, String> SIMPLIFY = ELEMENT.simplifyString;
```

Comment:
Codacy found an issue: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://app.codacy.com/gh/jdi-testing/jdi-light/pullRequest?prid=6397461)